debian/tests/docker-in-lxd

```diff
@@ -165,22 +165,54 @@ lxc exec docker --env DEBIAN_FRONTEND=noninteractive -- apt-get full-upgrade -y
 lxc restart docker
 sleep 5
 
+if [ -n "${AUTOPKGTEST_TMP}" ]; then
+    # When running autopkgtest locally, the binary packages being
+    # tested are placed into the temporary autopkgtest directory, and
+    # then apt installs them from there.  In order to be able to
+    # install these packages, we must also copy them to the lxd
+    # container.  If we don't do this here, the docker.io package
+    # installed inside the container will come from the archive, which
+    # is not what we want.  This needs to be done after the "lxc
+    # restart" command because /tmp is emptied after the reboot.
+    autopkgtest_tmp_root=$(realpath "${AUTOPKGTEST_TMP}/../")
```

Comment:
This is relying on some assumptions about the parent directory of `AUTOPKGTEST_TMP` -- is it safe to assume those won't break in the future? :grimacing:

Maybe there's something clever that could be done with `apt-get indextargets` or something to look for `file://` references and figure out which one has the right `docker.io_*.deb` in it?
(I guess similar to the one that's done further up in this script -- maybe this one rolled into that one? :sweat_smile:)

Reply:
I think something like this might work:

```
apt-get indextargets | grep-dctrl -nsRepo-URI -FURI -e 'file://.*' -a -FRepo-URI -e 'file://.*' | sed -e 's!^file://!!'
```

(need to add dctrl-tools to test depends)

Reply:
You're right; I completely miss the excerpt that did the same thing (in a better way).  I've now updated the branch and reused the code from that part.